test/e2e/index.test.ts

```diff
@@ -350,5 +350,16 @@ describe('e2e', () => {
         await page.evaluate(() => (document.activeElement as HTMLInputElement).value),
       ).to.equal('0-1');
     });
+
+    // https://github.com/mui/mui-x/issues/5590
+    it('should allow to click a button in NoRowsOverlay', async () => {
```

Comment:
It's impossible to cover it with unit test, since firing click on a button will always work, even if the button is covered with another element and in real browser user won't be able to click on the button